### PR TITLE
DOCSP-42819: add coding standards workflow

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

JIRA - <https://jira.mongodb.org/browse/DOCSP-42819>

Adapted the actions from the source driver repo but changed the rules for when the action should be run and on which filetypes

### CODE REVIEW

File: .github/workflows/coding-standards.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: "Coding Standards"
+
+on:
+  pull_request:
+    paths:
+      - 'source/**/*.php'
```
Comment (GromNaN) [inline .github/workflows/coding-standards.yml:6]: Add the .github directory so the GHA is run for this PR.
```suggestion
      - '.github/workflows/*.yml'
```

File: .github/workflows/coding-standards.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        uses: "actions/checkout@v4"
+
+      - name: "Setup"
+        uses: "./.github/actions/setup"
+        with:
+          php-version: ${{ env.PHP_VERSION }}
+          driver-version: ${{ env.DRIVER_VERSION }}
+
+      - name: "Run Rector"
+        run: "vendor/bin/rector --ansi --dry-run"
```
Comment (GromNaN) [inline .github/workflows/coding-standards.yml:48]: Rector is not setup for this project. You can remove this.

**Review by GromNaN — COMMENTED** [review]
You need to add a composer.json to install the tools:

```
    "require": {
        "mongodb/mongodb": "^1.20"
    },
    "require-dev": {
        "doctrine/coding-standard": "^12.0",
        "squizlabs/php_codesniffer": "^3.7",
        "vimeo/psalm": "^5.13"
    }
}
```

- the `phpcs.xml` inspired by https://github.com/mongodb/mongo-php-library/blob/master/phpcs.xml.dist
- the `psalm.xml` inspired by https://github.com/mongodb/mongo-php-library/blob/master/psalm.xml.dist

File: .github/workflows/static-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        with:
+          ref: ${{ github.event_name == 'workflow_dispatch' && inputs.ref || github.ref }}
+
+      - name: "Get SHA hash of checked out ref"
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        run: |
+          echo CHECKED_OUT_SHA=$(git rev-parse HEAD) >> $GITHUB_ENV
+
+      - name: "Setup"
+        uses: "./.github/actions/setup"
```
Comment (GromNaN) [inline .github/workflows/static-analysis.yml:38]: This file need to be imported in the project: https://github.com/mongodb/mongo-php-library/blob/v1.19/.github/actions/setup/action.yml

Comment (rustagir) [reply .github/workflows/static-analysis.yml:38]: It is part of this PR, fixed a few min ago

**Review by GromNaN — APPROVED** [review]
I've added the phpcs and psalm rules to make the tests pass.

**Pull request merged**